### PR TITLE
refactor: swap urls and pathbuilders location

### PR DESCRIPTION
**Pull request by aomi**
`urls.ts` is needed for client but `pathBuilders.ts` is only used for development. This PR just swaps the location and the references. Rational for this is we don't need build the library with any dev tools inside the codebase.

### CODE REVIEW

File: src/dev-tools/cli.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import fs from 'fs';
 import got from 'got';
 import yargs from 'yargs/yargs';
+import { detailedClassInformationUrl } from '../common/urls';
```
Comment (isaaccormack) [inline src/dev-tools/cli.ts:4]: can we move this import statement back to where it was?

File: src/dev-tools/section-dump.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import cheerio from 'cheerio';
 import fs from 'fs';
 import got from 'got';
+import { detailedClassInformationUrl } from '../common/urls';
```
Comment (isaaccormack) [inline src/dev-tools/section-dump.ts:4]: same with this import statement?

File: src/example-scraper/__tests__/example.test.ts [hunk 3]
```diff
@@ -1,9 +1,8 @@
 import * as cheerio from 'cheerio';
 import fs from 'fs';
 import each from 'jest-each';
-
-import { getScheduleFileByCourse, getSchedulePathsBySubject } from '../../common/pathBuilders';
-import { assertPageTitle } from "../../common/assertions";
+import { assertPageTitle } from '../../common/assertions';
```
Comment (isaaccormack) [inline src/example-scraper/__tests__/example.test.ts:4]: can we leave the empty line that was here between external and internal dependencies?

File: src/pages/courseListingTerm/__tests__/extractor.test.ts [hunk 4]
```diff
@@ -2,9 +2,9 @@ import appRoot from 'app-root-path';
 import * as cheerio from 'cheerio';
 import fs from 'fs';
 import path from 'path';
+import { getScheduleFileByCourse } from '../../../dev-tools/pathBuilders';
```
Comment (isaaccormack) [inline src/pages/courseListingTerm/__tests__/extractor.test.ts:5]: move this?

File: src/pages/detailedClassInformation/__tests__/index.test.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import * as cheerio from 'cheerio';
+import { getScheduleFileByCourse, getSectionFileByCRN } from '../../../dev-tools/pathBuilders';
```
Comment (isaaccormack) [inline src/pages/detailedClassInformation/__tests__/index.test.ts:2]: and this?

File: src/pages/searchForCourses/__tests__/extractor.test.ts [hunk 6]
```diff
@@ -2,9 +2,9 @@ import appRoot from 'app-root-path';
 import * as cheerio from 'cheerio';
 import fs from 'fs';
 import path from 'path';
+import { getScheduleFileByCourse } from '../../../dev-tools/pathBuilders';
```
Comment (isaaccormack) [inline src/pages/searchForCourses/__tests__/extractor.test.ts:5]: here too? idk maybe i'm being waaaaay to pedantic. I think I just configured my intellij to auto import this way

File: src/dev-tools/cli.ts (hunk 1 above)
Comment (aomi) [reply src/dev-tools/cli.ts:4]: I like this idea! I think there's a prettier or eslint feature that does that automatically?

Comment (aomi) [reply src/dev-tools/cli.ts:4]: https://github.com/benmosher/eslint-plugin-import ?

**Comment by isaaccormack** [comment]
From what i'm seeing, urls.ts is only used in dev-tools (ie. by cli.ts, schedule-dump.ts, and section-dump.ts).
Where are you seeing it used in the application code?

On the other hand, pathBuilders.ts is used by many app code tests, so maybe it should be left in common/ ?

**Comment by aomi** [comment]
@isaaccormack When the TypeScript project produces a ES5 library, we don't include any test-related things so keeping `pathBuilders` is kinda wasteful as it isn't needed in the actual functions the user accesses. My understanding was using `common` for things we want to include in the final JS build and `dev-tools` for development stuff.

**Comment by isaaccormack** [comment]
Ok that makes sense. I was thinking dev-tools was only for code in the tools related to development but it would be much easier to just put any code related to development in there because theres lots of re-use between testing code and dev-tools code.

In that case, maybe we should rename `dev-tools/` -> `dev/`
Then probably rename `pathBuilders.ts` -> `path-builders.ts` to fit the theme of file names in `dev/`
And while we're at it why don't we rename `example-scraper/` to just `example/`, I think thats self explanatory

**Review by isaaccormack — APPROVED** [review]
Nice! looks good. Thanks for making those changes. Looks like on commit 83d2d53 dev-tools wasn't actually renamed so I did that.

**Pull request merged**